modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/SystemDataSets.tsx

```diff
@@ -21,20 +27,121 @@ import openDefaultFailureToast from './utils/openDefaultFailureToast';
 import openDefaultSuccessToast from './utils/openDefaultSuccessToast';
 import {IDataSet, ISystemDataSet} from './utils/types';
 
+interface IFrontendDataSetContext {
+	onSelect: Function;
+	selectItems: Function;
+	selectable: boolean;
+	selectedItemsKey: keyof ISystemDataSet;
+	selectedItemsValue: Array<any>;
+}
+
+const SystemDataSetsView = ({
```

Comment:
If we are replacing the whole view, do we even need `FrontendDataSet`? Could this be directly a component in `<ClayModal.Body>`?

Reply:
`FrontendDataSet` still provides all the logic for the selection item handling, search, and loading spinner so it's still needed so it wouldn't need to be re-implemented.